diff --git a/torch/distributions/transforms.py b/torch/distributions/transforms.py
index 05e91a6feea..3a8e3d395a2 100644
--- a/torch/distributions/transforms.py
+++ b/torch/distributions/transforms.py
@@ -185,6 +185,16 @@ class Transform(object):
     def __repr__(self):
         return self.__class__.__name__ + '()'
 
+    def __getstate__(self):
+        # Do not pickle the cached inverse transform.
+        state = self.__dict__.copy()
+        state['_inv'] = None
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+        self._inv = None
+
     def forward_shape(self, shape):
         """
         Infers the shape of the forward computation, given the input shape.
