diff --git a/torch/_inductor/ir.py b/torch/_inductor/ir.py
index 386734212f3..5d66bde4706 100644
--- a/torch/_inductor/ir.py
+++ b/torch/_inductor/ir.py
@@ -3003,7 +3003,7 @@ class SliceView(View):
         sizevars = V.graph.sizevars
         dim_size = x.get_size()[dim]
 
-        if any(free_unbacked_symbols(x) for x in (start, end, dim_size)):
+        if any(free_unbacked_symbols(y) for y in (start, end, dim_size)):
 
             def clamp(x, lower, upper):  # type: ignore[no-untyped-def]
                 return sympy.Min(sympy.Max(x, lower), upper)
diff --git a/torch/fx/passes/runtime_assert.py b/torch/fx/passes/runtime_assert.py
index 9dd87d49ef5..4a4ac75d228 100644
--- a/torch/fx/passes/runtime_assert.py
+++ b/torch/fx/passes/runtime_assert.py
@@ -172,7 +172,7 @@ def insert_deferred_runtime_asserts(
             node.args,
         )
         try:
-            node.meta[val_key] = node.target(*fake_args)  # type: ignore[operator]
+            node.meta[val_key] = node.target(*fake_args, **node.kwargs)  # type: ignore[operator]
         except NotImplementedError:
             # This can happen when attempting to reify a symbol with an unsupported call_function node,
             # e.g. with NestedTensors + sym_size.int via match_symbol().
