diff --git a/torch/_prims/__init__.py b/torch/_prims/__init__.py
index 603658ea615..5c823458cba 100644
--- a/torch/_prims/__init__.py
+++ b/torch/_prims/__init__.py
@@ -308,7 +308,8 @@ def _make_prim(
     # register non-functional ops with old custom ops API
     cpp_schema = torch._C.parse_schema(name + schema)
     if use_old_custom_ops_api or not is_functional_schema(cpp_schema):
-        prim.define(name + schema, tags=torch.Tag.pt2_compliant_tag)
+        prim.define(name + schema, tags=tags or torch.Tag.pt2_compliant_tag)
+
         prim_impl.impl(name, _prim_impl)
         prim_autograd_impl.impl(name, _autograd_impl)
         prim_meta_impl.impl(name, meta)
