diff --git a/torch/__init__.py b/torch/__init__.py
index d7d861ba515..c8fdea5310b 100644
--- a/torch/__init__.py
+++ b/torch/__init__.py
@@ -750,17 +750,18 @@ def set_default_tensor_type(t):
 def set_default_dtype(d):
     r"""
 
-    Sets the default floating point dtype to :attr:`d`. Supports torch.float32
-    and torch.float64 as inputs. Other dtypes may be accepted without complaint
-    but are not supported and are unlikely to work as expected.
+    Sets the default floating point dtype to :attr:`d`. Supports torch.float16,
+    torch.float32, and torch.float64 as inputs. Other dtypes may be accepted
+    without complaint but are not supported and are unlikely to work as expected.
 
     When PyTorch is initialized its default floating point dtype is torch.float32,
     and the intent of set_default_dtype(torch.float64) is to facilitate NumPy-like
     type inference. The default floating point dtype is used to:
 
     1. Implicitly determine the default complex dtype. When the default floating point
-       type is float32 the default complex dtype is complex64, and when the default
-       floating point type is float64 the default complex type is complex128.
+       type is float16 the default complex dtype is complex32, when it is float32 the
+       default complex dtype is complex64, and when it is float64 the default complex
+       type is complex128.
     2. Infer the dtype for tensors constructed using Python floats or complex Python
        numbers. See examples below.
     3. Determine the result of type promotion between bool and integer tensors and
@@ -768,7 +769,7 @@ def set_default_dtype(d):
 
     Args:
         d (:class:`torch.dtype`): the floating point dtype to make the default.
-                                  Either torch.float32 or torch.float64.
+                                  Either torch.float16, torch.float32, or torch.float64.
 
     Example:
         >>> # xdoctest: +SKIP("Other tests may have changed the default type. Can we reset it?")
diff --git a/torch/_refs/__init__.py b/torch/_refs/__init__.py
index fa571489bd7..e842cfe7136 100644
--- a/torch/_refs/__init__.py
+++ b/torch/_refs/__init__.py
@@ -6296,6 +6296,8 @@ def _infer_scalar_type(obj):
             return torch.cfloat
         elif default_dtype is torch.double:
             return torch.cdouble
+        elif default_dtype is torch.float16:
+            return torch.complex32
         else:
             raise RuntimeError("invalid default scalar type for complex")
     if isinstance(obj, torch.Tensor):
