diff --git a/torch/fx/graph.py b/torch/fx/graph.py
index 60c864b43b7..4c607259960 100644
--- a/torch/fx/graph.py
+++ b/torch/fx/graph.py
@@ -378,7 +378,8 @@ class CodeGen:
         self, nodes, root_module: str, namespace: _Namespace, *,
         verbose: bool = False, include_stride: bool = False, include_device: bool = False, colored: bool = False
     ) -> PythonCode:
-        free_vars: List[str] = []
+        used_vars: Set[str] = {node.name for node in nodes if node.users}
+        free_vars: List[str] = [var for var in free_vars if var in used_vars]
         body: List[str] = []
         globals_: Dict[str, Any] = {}
         wrapped_fns: Dict[str, None] = {}
