diff --git a/torch/_dynamo/variables/builtin.py b/torch/_dynamo/variables/builtin.py
index 15bffbb5935..dd2e31b4059 100644
--- a/torch/_dynamo/variables/builtin.py
+++ b/torch/_dynamo/variables/builtin.py
@@ -526,6 +526,11 @@ class BuiltinVariable(VariableTracker):
                 and isinstance(args[0], ConstDictVariable)
                 and isinstance(args[1], variables.TensorVariable)
             )
+            and not (
+                isinstance(args[0], variables.ConstantVariable)
+                and isinstance(args[1], variables.TensorVariable)
+            )
+
         ):
             try:
                 fn = self.fn
