diff --git a/torch/_tensor.py b/torch/_tensor.py
index 2c47e6f53de..e98dc71d7fd 100644
--- a/torch/_tensor.py
+++ b/torch/_tensor.py
@@ -78,7 +78,8 @@ def _rebuild_from_type_v2(func, new_type, args, state):
 # NB: If you add a new method to Tensor, you must update
 # torch/__init__.py.in to add a type annotation for your method;
 # otherwise, it will not show up in autocomplete.
-class Tensor(torch._C._TensorBase):
+class Tensor(torch._C._TensorBase, torch._C._SymInt):
+
     def __deepcopy__(self, memo):
         if has_torch_function_unary(self):
             return handle_torch_function(Tensor.__deepcopy__, (self,), self, memo)
