diff --git a/torch/_library/utils.py b/torch/_library/utils.py
index 4e66a25406c..29774a85602 100644
--- a/torch/_library/utils.py
+++ b/torch/_library/utils.py
@@ -386,7 +386,7 @@ class MutationChecker:
             for a in self.flat_args
         ]
         was_mutated = [
-            not torch.equal(pre, post)
+            not (torch.equal(pre, post) or (torch.isnan(pre) and torch.isnan(post)))
             if isinstance(pre, torch.Tensor) and isinstance(post, torch.Tensor)
             else None
             for pre, post in zip(self.real_pre_hashes, real_post_hashes)
