diff --git a/torch/_dynamo/variables/user_defined.py b/torch/_dynamo/variables/user_defined.py
index ef80a55dd02..c8cac73c5df 100644
--- a/torch/_dynamo/variables/user_defined.py
+++ b/torch/_dynamo/variables/user_defined.py
@@ -2758,6 +2758,8 @@ class MutableMappingVariable(UserDefinedObjectVariable):
                 self.value.__dict__, AttrSource(self.source, "__dict__")
             )
             return self.generic_dict_vt
+        elif name in self.value.__dict__:
+            return variables.ConstantVariable(self.value.__dict__[name])
         elif out := self.generic_dict_vt.maybe_getitem_const(
             variables.ConstantVariable(name)
         ):
