diff --git a/torch/nn/modules/transformer.py b/torch/nn/modules/transformer.py
index dd699537e08..d0014f3d70e 100644
--- a/torch/nn/modules/transformer.py
+++ b/torch/nn/modules/transformer.py
@@ -464,7 +464,7 @@ class TransformerEncoder(Module):
             why_not_sparsity_fast_path = "src_key_padding_mask was None"
         elif (
             (not hasattr(self, "mask_check")) or self.mask_check
-        ) and not torch._nested_tensor_from_mask_left_aligned(
+        ) and not torch._dynamo.is_compiling() and not torch._nested_tensor_from_mask_left_aligned(
             src, src_key_padding_mask.logical_not()
         ):
             why_not_sparsity_fast_path = "mask_check enabled, and src and src_key_padding_mask was not left aligned"
