diff --git a/torch/_dynamo/variables/builder.py b/torch/_dynamo/variables/builder.py
index 86e40908f46..780ad8df26c 100644
--- a/torch/_dynamo/variables/builder.py
+++ b/torch/_dynamo/variables/builder.py
@@ -2881,6 +2881,10 @@ def handle_traced_output(example_value, tx, proxy, options, subclass_type, targe
     import torch._utils
 
     if isinstance(example_value, torch.Tensor):
+        if example_value.layout is not torch.strided:
+            raise torch._dynamo.exc.Unsupported(
+                "non-strided tensors are not supported"
+            )
         var = construct_tensor_variable(
             target_cls, tx, proxy, example_value, subclass_type, options
         )
diff --git a/torch/_inductor/lowering.py b/torch/_inductor/lowering.py
index 7001fe6a66d..aab0b346ed6 100644
--- a/torch/_inductor/lowering.py
+++ b/torch/_inductor/lowering.py
@@ -2141,6 +2141,9 @@ def unsupported_input_tensor(t: torch.Tensor, node=None):
     if t.is_meta:
         return True
 
+    if t.is_sparse:
+        return True
+
     if t.dtype == torch.float8_e8m0fnu:
         if not node:
             return True
