diff --git a/torch/_inductor/codegen/cpp_gemm_template.py b/torch/_inductor/codegen/cpp_gemm_template.py
index 9942185d9e2..e93f75a50f0 100644
--- a/torch/_inductor/codegen/cpp_gemm_template.py
+++ b/torch/_inductor/codegen/cpp_gemm_template.py
@@ -294,7 +294,7 @@ class CppPackedGemmTemplate(CppTemplate):
 
         # we favor square-sized thread blocks for good data reuse
         def get_better_blocking(blocking, best_blocking):
-            if best_blocking is None:
+            if best_blocking is None or template_buffer_has_other_users:
                 best_blocking = blocking
             else:
                 block_m_size = blocking.block_m * register_blocking.block_m
