diff --git a/torch/_dynamo/variables/builtin.py b/torch/_dynamo/variables/builtin.py
index 5a19e707689..8dbd2ae4f90 100644
--- a/torch/_dynamo/variables/builtin.py
+++ b/torch/_dynamo/variables/builtin.py
@@ -2270,8 +2270,8 @@ class BuiltinVariable(VariableTracker):
                 ),
                 sym_num=None,
             )
-        if hasattr(a, "set_items") and hasattr(b, "set_items"):
-            return SetVariable(list(a.set_items | b.set_items))
+        if isinstance(a, ConstDictVariable) and isinstance(b, ConstDictVariable):
+            return ConstDictVariable(a.items | b.items)
         # None no-ops this handler and lets the driving function proceed
         return None
 
diff --git a/torch/_dynamo/variables/dicts.py b/torch/_dynamo/variables/dicts.py
index 2703d8c4eb7..cbbac98be57 100644
--- a/torch/_dynamo/variables/dicts.py
+++ b/torch/_dynamo/variables/dicts.py
@@ -534,8 +534,12 @@ class ConstDictVariable(VariableTracker):
             self.items.pop(key)
             self.items[key] = val
             return ConstantVariable.create(None)
-        else:
-            return super().call_method(tx, name, args, kwargs)
+        elif name == "__or__":
+            assert len(args) == 1 and not kwargs
+            other = args[0]
+            if isinstance(other, ConstDictVariable):
+                return ConstDictVariable(self.items | other.items)
+        return super().call_method(tx, name, args, kwargs)
 
     def unpack_var_sequence(self, tx):
         self.install_dict_keys_match_guard()
