diff --git a/torch/_inductor/lowering.py b/torch/_inductor/lowering.py
index 1cb970f4446..557116bf3f5 100644
--- a/torch/_inductor/lowering.py
+++ b/torch/_inductor/lowering.py
@@ -3432,6 +3432,9 @@ def _adaptive_avg_pool2d(x, output_size):
 
     h_out, w_out = output_size
 
+    if h_out == 0 or w_out == 0:
+        return fallback_adaptive_avg_pool2d(x, output_size)
+
     # no-op if the same input and output
     if h_in == h_out and w_in == w_out:
         return clone(x)
