diff --git a/torch/_dynamo/symbolic_convert.py b/torch/_dynamo/symbolic_convert.py
index a15eb092b2b..5a12cb96d01 100644
--- a/torch/_dynamo/symbolic_convert.py
+++ b/torch/_dynamo/symbolic_convert.py
@@ -3261,7 +3261,7 @@ class InliningInstructionTranslator(InstructionTranslatorBase):
             fglobals_vt = self.output.side_effects.track_object_existing(
                 fglobals_value, fglobals_vt
             )
-            self.output.side_effects.store_attr(fglobals_vt, name, value)
+            self.output.side_effects.store_global(fglobals_vt, name, value)
 
 
 class InliningGeneratorInstructionTranslator(InliningInstructionTranslator):
diff --git a/torch/_dynamo/variables/builder.py b/torch/_dynamo/variables/builder.py
index cf7ec28a347..11bf3180cc7 100644
--- a/torch/_dynamo/variables/builder.py
+++ b/torch/_dynamo/variables/builder.py
@@ -532,7 +532,7 @@ class VariableBuilder:
             }
             return ConstDictVariable(result, type(value))
         elif value is sys.modules:
-            self.install_guards(GuardBuilder.FUNCTION_MATCH)
+            self.install_guards(GuardBuilder.GLOBAL_MATCH)
             return PythonSysModulesVariable(source=self.source)
         elif CustomizedDictVariable.is_matching_cls_hf(type(value)):
             self.install_guards(GuardBuilder.TYPE_MATCH)
