diff --git a/torch/jit/_recursive.py b/torch/jit/_recursive.py
index a466ff8aee4..bec55409e2a 100644
--- a/torch/jit/_recursive.py
+++ b/torch/jit/_recursive.py
@@ -198,7 +198,7 @@ def infer_concrete_type_builder(nn_module, share_types=True):
         added_names.add(name)
 
     # populate constants_set
-    constants_set = getattr(nn_module, "__constants__", set())
+    constants_set = set(getattr(nn_module, "__constants__", set()))
 
     # Constants annotated via `Final[T]` rather than being added to `__constants__`
     for name, ann in class_annotations.items():
