diff --git a/torch/_inductor/sizevars.py b/torch/_inductor/sizevars.py
index 44689734d80..a6065c938bf 100644
--- a/torch/_inductor/sizevars.py
+++ b/torch/_inductor/sizevars.py
@@ -908,7 +908,7 @@ class SizeVarAllocator:
         return expr
 
     def atomically_apply_size_hint(
-        self, expr: Union[Expr, int], *, fallback: Optional[int] = None
+        self, expr: Union[Expr, int], *, fallback: Optional[int] = None, hint_override: Optional[dict] = None
     ) -> Union[Expr, int]:
         if isinstance(expr, (int, sympy.Integer)):
             return int(expr)
@@ -925,10 +925,10 @@ class SizeVarAllocator:
         assert isinstance(expr, Expr), type(expr)
         free_symbols = expr.free_symbols
         size_dict = {
-            symbol: V.graph.sizevars.size_hint(symbol, fallback=fallback)
+            symbol: V.graph.sizevars.size_hint(symbol, fallback=fallback, hint_override=hint_override)
             for symbol in free_symbols
         }
-        return expr.subs(size_dict)
+        return int(expr.subs(size_dict))
 
     def offset_var(self, index: Expr, vars: Sequence[sympy.Symbol]) -> Expr:
         """Extract offset part of an indexing expression"""
