diff --git a/torch/utils/_sympy/solve.py b/torch/utils/_sympy/solve.py
index cdc722d20fa..2a1d9280ffc 100644
--- a/torch/utils/_sympy/solve.py
+++ b/torch/utils/_sympy/solve.py
@@ -115,7 +115,7 @@ def _try_isolate_lhs(
 
         # If we can't tell whether 'other' is negative or positive, we do nothing.
         # That is because we don't know whether we have mirror the operation or not.
-        if not (isinstance(e, INEQUALITY_TYPES) and other.is_negative is None):
+        if other.is_nonzero and not (isinstance(e, INEQUALITY_TYPES) and other.is_negative is None):
             # Divide both sides by 'other'.
             lhs = lhs / other
             rhs = rhs / other
