diff --git a/torch/_export/passes/_node_metadata_hook.py b/torch/_export/passes/_node_metadata_hook.py
index 9ca554349fa..de1eebcda9c 100644
--- a/torch/_export/passes/_node_metadata_hook.py
+++ b/torch/_export/passes/_node_metadata_hook.py
@@ -37,7 +37,7 @@ def _node_metadata_hook(node: torch.fx.Node, stack_trace: str) -> None:
         node.meta["val"] = fake_res
 
     node.meta["stack_trace"] = stack_trace
-    node.meta["nn_module_stack"] = arg_meta.get("nn_module_stack", {})
+    node.meta["nn_module_stack"] = arg_meta.get("nn_module_stack", [])
     node.meta["torch_fn"] = (
         f"{node.target.__name__}_0",
         f"{node.target.__class__.__name__}.{node.target.__name__}",
diff --git a/torch/export/unflatten.py b/torch/export/unflatten.py
index 31701d9fb68..943209cec36 100644
--- a/torch/export/unflatten.py
+++ b/torch/export/unflatten.py
@@ -892,7 +892,7 @@ class _ModuleFrame:
                 return node_idx
 
             node_module_stack = (
-                [path for path, ty in node.meta["nn_module_stack"].values()]
+                [path for path, ty in node.meta["nn_module_stack"]]
                 if "nn_module_stack" in node.meta
                 else self.module_stack
             )
