diff --git a/torch/_dynamo/source.py b/torch/_dynamo/source.py
index 02ae0bfb209..dd7fc1ec68b 100644
--- a/torch/_dynamo/source.py
+++ b/torch/_dynamo/source.py
@@ -592,6 +592,15 @@ class FloatTensorSource(ChainedSource):
         return self.base.guard_source()
 
 
+@dataclasses.dataclass(frozen=True)
+class SubclassAttrListSource(ChainedSource):
+    def name(self) -> str:
+        return f"___subclass_attr_list({self.base.name()})"
+
+    def guard_source(self):
+        return self.base.guard_source()
+
+
 @dataclasses.dataclass(frozen=True)
 class CallMethodItemSource(ChainedSource):
     def name(self) -> str:
