diff --git a/torch/_meta_registrations.py b/torch/_meta_registrations.py
index 86685968ac9..816478898ae 100644
--- a/torch/_meta_registrations.py
+++ b/torch/_meta_registrations.py
@@ -5355,7 +5355,7 @@ def meta__flash_attention_forward(
     scale: Optional[float] = None,
 ):
     batch_size = query.size(0)
-    max_seqlen_batch_q = query.size(1)
+    max_seqlen_batch_q = query.size(1) if query.dim() > 1 else 1
     num_heads = query.size(2)
     head_dim = query.size(3)
 
diff --git a/torch/nested/_internal/sdpa.py b/torch/nested/_internal/sdpa.py
index 5a2ce0ada12..8c3924d5810 100644
--- a/torch/nested/_internal/sdpa.py
+++ b/torch/nested/_internal/sdpa.py
@@ -604,7 +604,7 @@ def _pad_last_dim(
 
 # TODO: coalesce with torch/nn/utils/attention.py
 def _calculate_scale(query, scale):
-    softmax_scale = scale if scale is not None else math.sqrt(1.0 / query.size(-1))
+    softmax_scale = scale if scale is not None else 1.0 / math.sqrt(query.size(-1))
     return softmax_scale
 
 
