diff --git a/torch/_dynamo/utils.py b/torch/_dynamo/utils.py
index 41131be1055..df15b11fbda 100644
--- a/torch/_dynamo/utils.py
+++ b/torch/_dynamo/utils.py
@@ -1751,6 +1751,8 @@ def get_fake_value(node, tx, allow_non_graph_fake=False):
         UserErrorType,
     )
 
+    if node.op == "placeholder":
+        return tracer.real_value_cache.get(node, None)
     op = node.op
 
     # FX Node should always return the same fake value
