diff --git a/torch/_inductor/standalone_compile.py b/torch/_inductor/standalone_compile.py
index 6a1d17dbe9a..0ba9e6d9e02 100644
--- a/torch/_inductor/standalone_compile.py
+++ b/torch/_inductor/standalone_compile.py
@@ -11,7 +11,7 @@ from torch._dynamo.utils import dynamo_timed
 from torch._inductor.cudagraph_utils import BoxedDeviceIndex
 from torch._inductor.runtime.cache_dir_utils import temporary_cache_dir
 from torch._inductor.utils import BoxedBool, InputType
-from torch._subclasses import FakeTensorMode
+from torch._subclasses.fake_tensor import FakeTensorMode
 from torch.fx.experimental.symbolic_shapes import ShapeEnv
 
 from . import config
diff --git a/torch/_inductor/utils.py b/torch/_inductor/utils.py
index be1c2202aba..7a77bb57e1a 100644
--- a/torch/_inductor/utils.py
+++ b/torch/_inductor/utils.py
@@ -51,7 +51,7 @@ import sympy
 
 import torch
 from torch._inductor.runtime.hints import DeviceProperties
-from torch.fx.experimental.symbolic_shapes import ShapeEnv
+from torch.fx.experimental.symbolic_shapes import ShapeEnv, ShapeGuard
 from torch.utils._ordered_set import OrderedSet
 from torch.utils._pytree import tree_map_only
 
