diff --git a/torch/_inductor/codegen/triton.py b/torch/_inductor/codegen/triton.py
index 5b39a0fe435..2a57ca90e09 100644
--- a/torch/_inductor/codegen/triton.py
+++ b/torch/_inductor/codegen/triton.py
@@ -216,7 +216,7 @@ class BlockPtrOptions:
         return V.kernel.index_to_str(advance)
 
     def has_rindex(self):
-        return "RBLOCK" in self.block_shape
+        return "roffset" in self.offsets
 
     def has_tmpmask(self):
         return False  # block_ptr can't do indirect indexing
