diff --git a/torch/_prims_common/wrappers.py b/torch/_prims_common/wrappers.py
index 990fdf60cb1..078bc712a15 100644
--- a/torch/_prims_common/wrappers.py
+++ b/torch/_prims_common/wrappers.py
@@ -133,7 +133,7 @@ class elementwise_type_promotion_wrapper:
 
             result = fn(**bound.arguments)
 
-            if isinstance(result, TensorLike):
+            if isinstance(result, (TensorLike, torch.Tensor)):
                 return _maybe_convert_to_dtype(result, result_dtype)
             if isinstance(result, Sequence):
                 return tuple(_maybe_convert_to_dtype(x, result_dtype) for x in result)
