diff --git a/torch/_dynamo/variables/builder.py b/torch/_dynamo/variables/builder.py
index f7ac4d073b5..581001fe437 100644
--- a/torch/_dynamo/variables/builder.py
+++ b/torch/_dynamo/variables/builder.py
@@ -550,7 +550,10 @@ class VariableBuilder:
         # Note - There are some nested values where types mismatch!
         # We want to get those out and wrap those.
         if is_function_or_wrapper(value):
+            prior = value
             value = inspect.getattr_static(value, "_torchdynamo_inline", value)
+            if value is not prior:
+                self.source = AttrSource(self.source, "_torchdynamo_inline")
 
         # Everything else (NB: order matters!)
         if is_traceable_wrapper_subclass(value) or istype(
