diff --git a/torch/_dynamo/variables/builder.py b/torch/_dynamo/variables/builder.py
index f7ac4d073b5..81717e73306 100644
--- a/torch/_dynamo/variables/builder.py
+++ b/torch/_dynamo/variables/builder.py
@@ -550,7 +550,7 @@ class VariableBuilder:
         # Note - There are some nested values where types mismatch!
         # We want to get those out and wrap those.
         if is_function_or_wrapper(value):
-            value = inspect.getattr_static(value, "_torchdynamo_inline", value)
+            value = inspect.unwrap(value, "_torchdynamo_inline", value)
 
         # Everything else (NB: order matters!)
         if is_traceable_wrapper_subclass(value) or istype(
diff --git a/torch/_dynamo/variables/functions.py b/torch/_dynamo/variables/functions.py
index 90c2d0665ae..484749e39c8 100644
--- a/torch/_dynamo/variables/functions.py
+++ b/torch/_dynamo/variables/functions.py
@@ -153,7 +153,7 @@ class UserFunctionVariable(BaseUserFunctionVariable):
             fn, (types.FunctionType, torch.jit.ScriptFunction)
         ), f"expected FunctionType found {typestr(fn)} {fn}"
         # unpack @torch._dynamo.optimize()(fn) wrapped function
-        fn = inspect.getattr_static(fn, "_torchdynamo_inline", fn)
+        fn = inspect.unwrap(fn, "_torchdynamo_inline", fn)
         self.fn: types.FunctionType = fn
 
     def as_python_constant(self):
