diff --git a/torch/_dynamo/convert_frame.py b/torch/_dynamo/convert_frame.py
index 856e399a6d8..41b6adbcc39 100644
--- a/torch/_dynamo/convert_frame.py
+++ b/torch/_dynamo/convert_frame.py
@@ -648,7 +648,8 @@ def _compile(
             one_graph,
             export,
             export_constraints,
-            mutated_closure_cell_contents,
+            mutated_closure_cell_contents.copy(),
+
             frame_state=frame_state,
             speculation_log=speculation_log,
             distributed_state=distributed_state,
@@ -865,7 +866,8 @@ def _compile(
     ):
         restart_reasons: set[str] = set()
         # This is shared across restarts
-        mutated_closure_cell_contents: Set[str] = set()
+        mutated_closure_cell_contents: Set[str] = set()  # Track mutated closure cells
+
         speculation_log = SpeculationLog()
         if compile_pg := get_compile_pg():
             distributed_state = DistributedState(compile_pg, LocalState())
diff --git a/torch/_dynamo/symbolic_convert.py b/torch/_dynamo/symbolic_convert.py
index 40f05e94891..089a65a7161 100644
--- a/torch/_dynamo/symbolic_convert.py
+++ b/torch/_dynamo/symbolic_convert.py
@@ -2787,7 +2787,8 @@ class InstructionTranslator(InstructionTranslatorBase):
         one_graph,
         export,
         export_constraints,
-        mutated_closure_cell_contents: Set[str],
+        mutated_closure_cell_contents: Set[str],  # Track mutated closure cells
+
         frame_state,
         speculation_log: SpeculationLog,
         distributed_state: Optional[DistributedState],
diff --git a/torch/_dynamo/variables/functions.py b/torch/_dynamo/variables/functions.py
index 1e7d4ab49e0..9a94671a1d7 100644
--- a/torch/_dynamo/variables/functions.py
+++ b/torch/_dynamo/variables/functions.py
@@ -276,7 +276,8 @@ class UserFunctionVariable(BaseUserFunctionVariable):
 
                         if (
                             closure_cell_contents.name()
-                            not in tx.mutated_closure_cell_contents
+                            not in tx.mutated_closure_cell_contents.copy()
+
                         ):
                             # Optimistically don't allocate the cell, to
                             # reduce the number of side effects.  This is
