diff --git a/torch/nn/modules/module.py b/torch/nn/modules/module.py
index 2764dc27bb3..baad7a44690 100644
--- a/torch/nn/modules/module.py
+++ b/torch/nn/modules/module.py
@@ -451,7 +451,8 @@ class Module:
 
     training: bool
     _parameters: dict[str, Optional[Parameter]]
-    _buffers: dict[str, Optional[Tensor]]
+    _buffers: dict[str, Optional[Union[Tensor, Any]]]
+
     _non_persistent_buffers_set: set[str]
     _backward_pre_hooks: dict[int, Callable]
     _backward_hooks: dict[int, Callable]
