diff --git a/torch/_dynamo/variables/builder.py b/torch/_dynamo/variables/builder.py
index d62dbe6d1f3..b49dda1fda2 100644
--- a/torch/_dynamo/variables/builder.py
+++ b/torch/_dynamo/variables/builder.py
@@ -1350,7 +1350,13 @@ class VariableBuilder:
         # Get all the values from the range iterator; no need to install guards
         # on items since `RANGE_ITERATOR_MATCH` guarantees the same items.
         items = [ConstantVariable.create(v) for v in copy.deepcopy(value)]
-        return ListIteratorVariable(items, mutation_type=ValueMutationNew())
+        result = TupleIteratorVariable(
+            TupleVariable(items),
+            mutation_type=ValueMutationNew(),
+            source=self.source,
+        )
+        return self.set_source_and_track_mutable(value, result)
+
 
     def wrap_slice_range(self, value: Union[slice, range]):
         items = [
