diff --git a/torch/utils/_sympy/functions.py b/torch/utils/_sympy/functions.py
index 23d0bc24bf1..b139397f93a 100644
--- a/torch/utils/_sympy/functions.py
+++ b/torch/utils/_sympy/functions.py
@@ -310,7 +310,7 @@ class ModularIndexing(sympy.Function):
     def eval(
         cls, base: sympy.Integer, divisor: sympy.Integer, modulus: sympy.Integer
     ) -> Optional[sympy.Basic]:
-        if base == 0 or modulus == 1:
+        if base == 0 or divisor == 0 or modulus == 1:
             return sympy.S.Zero
 
         if (
