diff --git a/torch/_decomp/decompositions.py b/torch/_decomp/decompositions.py
index 634c4b6b495..8e48fd991e2 100644
--- a/torch/_decomp/decompositions.py
+++ b/torch/_decomp/decompositions.py
@@ -5066,7 +5066,7 @@ def scaled_dot_product_flash_attention_for_cpu(
         lambda: "q, k, v should have the same head size",
     )
 
-    output, attn = aten._scaled_dot_product_attention_math.default(
+    output, attn = aten._scaled_dot_product_attention_math(
         query,
         key,
         value,
