diff --git a/torch/fx/passes/runtime_assert.py b/torch/fx/passes/runtime_assert.py
index f8c12327f31..5989c724c48 100644
--- a/torch/fx/passes/runtime_assert.py
+++ b/torch/fx/passes/runtime_assert.py
@@ -336,10 +336,6 @@ def insert_deferred_runtime_asserts(
                 if (
                     node.args[0] == True  # noqa: E712
                     or (assert_expr := _get_sym_val(node.args[0])) in expr_to_proxy
-                    or (
-                        assert_expr is not None
-                        and _is_bound_expr_for_symbol(assert_expr)
-                    )
                 ):
                     arg = node.args[0]
                     gm.graph.erase_node(node)
@@ -348,6 +344,7 @@ def insert_deferred_runtime_asserts(
                 else:
                     added_asserts.add(assert_expr)  # type: ignore[arg-type]
 
+
             # hash cons, replace function calls that return torch.SymInts with direct references to
             # FX nodes built up to reify the sympy expression.
             if (
