diff --git a/torch/_dynamo/variables/misc.py b/torch/_dynamo/variables/misc.py
index c6a2124c871..aa4c3381164 100644
--- a/torch/_dynamo/variables/misc.py
+++ b/torch/_dynamo/variables/misc.py
@@ -298,7 +298,7 @@ class SuperVariable(VariableTracker):
                     tx_old
                 )
 
-        unimplemented(f"non-function or method super: {inner_fn}")
+        return unimplemented(f"non-function or method super: {inner_fn}")
 
 
 class ExceptionVariable(VariableTracker):
diff --git a/torch/_dynamo/variables/torch_function.py b/torch/_dynamo/variables/torch_function.py
index 3946dccd8dc..f0de5756283 100644
--- a/torch/_dynamo/variables/torch_function.py
+++ b/torch/_dynamo/variables/torch_function.py
@@ -597,9 +597,8 @@ class TensorWithTFOverrideVariable(TensorVariable):
 
         # This simulates shallow-copying the tensor object.
         kwargs = dict(tensor_var.__dict__)
-        assert kwargs.pop("class_type") is torch.Tensor, (
-            "invalid class type in TensorWithTFOverrideVariable.from_tensor_var"
-        )
+        if kwargs.pop("class_type") is not torch.Tensor:
+            raise TypeError("invalid class type in TensorWithTFOverrideVariable.from_tensor_var")
         torch_fn_var = build_torch_function_fn(tx, class_type, cls_source)
         var = cls(torch_function_fn=torch_fn_var, class_type=class_type, **kwargs)
         var.install_global(tx)
@@ -640,7 +639,7 @@ class TensorWithTFOverrideVariable(TensorVariable):
 
         if hasattr(torch.Tensor, name):
             if _is_attr_overidden(tx, self, name):
-                unimplemented(
+                return unimplemented(
                     f"Accessing overridden method/attribute {name} on a tensor"
                     " subclass with a __torch_function__ override is not supported"
                 )
diff --git a/torch/_dynamo/variables/user_defined.py b/torch/_dynamo/variables/user_defined.py
index b842a552649..9bcd80532e8 100644
--- a/torch/_dynamo/variables/user_defined.py
+++ b/torch/_dynamo/variables/user_defined.py
@@ -578,7 +578,7 @@ class UserDefinedClassVariable(UserDefinedVariable):
 
                 assert all(x is not None for x in items)
 
-            return variables.NamedTupleVariable(items, self.value)
+            return variables.NamedTupleVariable(items, self.value, **var_tracker_kwargs)
         elif is_frozen_dataclass(self.value) and self.is_standard_new():
             fields = dataclasses.fields(self.value)
             items = list(args)
