diff --git a/torch/fx/experimental/normalize.py b/torch/fx/experimental/normalize.py
index d7bd2c11468..51b038236af 100644
--- a/torch/fx/experimental/normalize.py
+++ b/torch/fx/experimental/normalize.py
@@ -45,9 +45,10 @@ class NormalizeArgs(Transformer):
 
         def get_type(arg):
             if isinstance(arg, fx.Node):
-                return n.meta["type"] if "type" in n.meta else None
+                return arg.meta["type"] if "type" in arg.meta else None
             return type(arg)
 
+
         arg_types = map_aggregate(n.args, get_type)
         assert isinstance(arg_types, tuple)
         arg_types = tuple([create_type_hint(i) for i in arg_types])
