diff --git a/torch/_inductor/codegen/cpp_wrapper_cpu.py b/torch/_inductor/codegen/cpp_wrapper_cpu.py
index 1db5bcdd39f..c6ca4eb2d12 100644
--- a/torch/_inductor/codegen/cpp_wrapper_cpu.py
+++ b/torch/_inductor/codegen/cpp_wrapper_cpu.py
@@ -192,8 +192,9 @@ class CppWrapperCpu(WrapperCodeGen):
         self.header.splice("typedef at::Half half;")
         self.header.splice("typedef at::BFloat16 bfloat16;")
         self.header.splice("#include <c10/util/generic_math.h>")
+        if config.cpp.enable_kernel_profile:
+            self.header.splice("#include <torch/csrc/profiler/events.h
 
-        if not V.graph.aot_mode:
             self.header.splice(
                 """
                 #include <pybind11/pybind11.h>
