diff --git a/torch/distributions/mixture_same_family.py b/torch/distributions/mixture_same_family.py
index b5a853797b4..72ede5dfe54 100644
--- a/torch/distributions/mixture_same_family.py
+++ b/torch/distributions/mixture_same_family.py
@@ -95,7 +95,8 @@ class MixtureSameFamily(Distribution):
         event_shape = self._component_distribution.event_shape
         self._event_ndims = len(event_shape)
         super().__init__(
-            batch_shape=cdbs, event_shape=event_shape, validate_args=validate_args
+            batch_shape=torch.broadcast_shapes(mdbs, cdbs), event_shape=event_shape, validate_args=validate_args
+
         )
 
     def expand(self, batch_shape, _instance=None):
