diff --git a/torch/_inductor/fx_passes/split_cat.py b/torch/_inductor/fx_passes/split_cat.py
index 7636d7cde36..8789f210212 100644
--- a/torch/_inductor/fx_passes/split_cat.py
+++ b/torch/_inductor/fx_passes/split_cat.py
@@ -1701,7 +1701,8 @@ def normalize_split_default_aten(match: Match, *args, **kwargs):
         return
     if split_dim < 0:  # Normalize split dim
         split_dim += split_input.meta["val"].dim()
-    split_section_list = [split_size] * (len(split_node.meta["val"]))
+    split_section_list = split_sections
+
     new_args = (split_input, split_section_list)
     new_kwargs = {"dim": split_dim}
     if (
