diff --git a/torch/export/unflatten.py b/torch/export/unflatten.py
index e95ba9e7b8c..c8010e87aac 100644
--- a/torch/export/unflatten.py
+++ b/torch/export/unflatten.py
@@ -1015,6 +1015,8 @@ class _ModuleFrame:
             with self.graph.inserting_after(None):
                 arg_nodes = [
                     self.graph.placeholder(f"_positional_arg_{idx}")
+                    if not isinstance(signature.inputs[idx], ConstantArgument)
+                    else signature.inputs[idx].value
                     for idx in range(args_spec.num_children)
                 ]
                 kwarg_nodes = {}
