diff --git a/torch/_dynamo/polyfills/__init__.py b/torch/_dynamo/polyfills/__init__.py
index 4b3d911e202..dec650983cc 100644
--- a/torch/_dynamo/polyfills/__init__.py
+++ b/torch/_dynamo/polyfills/__init__.py
@@ -227,7 +227,7 @@ def construct_dict(cls, /, *args, **kwargs):
         src = args[0]
 
         # Ensure that the overridden __iter__ method is invoked
-        if isinstance(src, (dict, MutableMapping)):
+        if isinstance(src, (dict, MutableMapping, types.MappingProxyType)):
             for key in src:
                 # This will inline the __getitem__ of the src object
                 dst[key] = src[key]
@@ -236,6 +236,7 @@ def construct_dict(cls, /, *args, **kwargs):
             for key, value in src:
                 dst[key] = value
 
+
     if kwargs:
         for key in kwargs:
             dst[key] = kwargs[key]
