diff --git a/torch/cuda/__init__.py b/torch/cuda/__init__.py
index f735a414ecc..c79175cb1b4 100644
--- a/torch/cuda/__init__.py
+++ b/torch/cuda/__init__.py
@@ -135,6 +135,8 @@ def is_bf16_supported(including_emulation: bool = True):
     if torch.version.hip:
         return True
 
+    if not torch.cuda.is_available():
+        return False
     device = torch.cuda.current_device()
 
     # Check for CUDA version and device compute capability.
