diff --git a/torch/_functorch/autograd_function.py b/torch/_functorch/autograd_function.py
index ec3a3f528ab..4b44209099f 100644
--- a/torch/_functorch/autograd_function.py
+++ b/torch/_functorch/autograd_function.py
@@ -820,7 +820,7 @@ class AutogradFunctionApply(HigherOrderOperator):
         class ApplyTemplate(torch.autograd.Function):
             @staticmethod
             # pyrefly: ignore [bad-override]
-            def forward(ctx: Any, *args: Any) -> Any:
+            def forward(*args: Any) -> Any:
                 nonlocal saved_values
 
                 # The Interpreter here is required to propagate metadata
@@ -839,6 +839,10 @@ class AutogradFunctionApply(HigherOrderOperator):
                         for proxy in _get_proxies(t):
                             proxy.node.meta["saved_tensor_with_no_vc_check"] = True
 
+                return output
+
+            @staticmethod
+            def setup_context(ctx: Any, inputs: Any, output: Any) -> None:
                 # If users call ctx.mark_non_differentiable() in the original fwd function.
                 if len(non_differentiable_idx) > 0:
                     non_differentiable_output = []
@@ -847,8 +851,6 @@ class AutogradFunctionApply(HigherOrderOperator):
                             non_differentiable_output.append(x)
                     ctx.mark_non_differentiable(*non_differentiable_output)
 
-                return output
-
             @staticmethod
             def backward(ctx: Any, *grad: Any) -> Any:
                 # The Interpreter here is required to propagate metadata
@@ -859,6 +861,7 @@ class AutogradFunctionApply(HigherOrderOperator):
                     raise AssertionError("saved_values must not be None")
                 return torch.fx.Interpreter(bwd).run(*grad, *saved_values)
 
+
         return ApplyTemplate.apply(*fwd_args)
 
 
