diff --git a/torch/_inductor/utils.py b/torch/_inductor/utils.py
index d48206f091f..f7689a380ab 100644
--- a/torch/_inductor/utils.py
+++ b/torch/_inductor/utils.py
@@ -799,15 +799,16 @@ def sympy_str(expr: sympy.Expr) -> str:
     if isinstance(expr, sympy.Symbol):
         return expr.name
     if isinstance(expr, sympy.Add):
-        return " + ".join(map(sympy_str, expr.args))
+        return " + ".join(sympy_str(arg) if not isinstance(arg, sympy.Mul) else f"({sympy_str(arg)})" for arg in expr.args)
     if isinstance(expr, sympy.Mul):
-        return " * ".join(map(sympy_str, expr.args))
+        return " * ".join(sympy_str(arg) if not isinstance(arg, sympy.Add) else f"({sympy_str(arg)})" for arg in expr.args)
 
     if isinstance(expr, (ModularIndexing, CleanDiv, FloorDiv, Identity)):
         return f"{expr.func.__name__}({', '.join(map(sympy_str, expr.args))})"
     return str(expr)
 
 
+
 def get_bounds_index_expr(index: sympy.Expr) -> ValueRanges[Any]:
     from .virtualized import V
 
