diff --git a/torch/_inductor/scheduler.py b/torch/_inductor/scheduler.py
index 05e3b1b6440..8f2d7100fc7 100644
--- a/torch/_inductor/scheduler.py
+++ b/torch/_inductor/scheduler.py
@@ -1923,7 +1923,21 @@ class FusedSchedulerNode(BaseSchedulerNode):
             # of the template node and its epilogue requires the same type of dependencies
             assert isinstance(node2.node, MultiOutput)
             assert len(node2.read_writes.writes) == 1
-            assert isinstance(next(iter(node2.read_writes.writes)), StarDep)
+            if isinstance(next(iter(node2.read_writes.writes)), StarDep):
+                name = next(iter(node2.read_writes.writes)).name
+                template_nodes = [node for node in node1.get_nodes() if node.is_template()]
+                assert len(template_nodes) == 1
+                template_node = template_nodes[0]
+                assert len(template_node.read_writes.writes) == 1
+                write = next(iter(template_node.read_writes.writes))
+                assert isinstance(write, MemoryDep)
+                node2.read_writes.writes = OrderedSet(
+                    [
+                        MemoryDep(
+                            name, write.index, write.var_names, write.size, write.mode
+                        ),
+                    ]
+                )
             name = next(iter(node2.read_writes.writes)).name
             template_nodes = [node for node in node1.get_nodes() if node.is_template()]
             assert len(template_nodes) == 1
