diff --git a/torch/fx/experimental/sym_node.py b/torch/fx/experimental/sym_node.py
index 268f56e3214..c126ae0568d 100644
--- a/torch/fx/experimental/sym_node.py
+++ b/torch/fx/experimental/sym_node.py
@@ -1095,7 +1095,9 @@ def _make_node_magic(method, func):
         except Exception:
             log.warning("failed to eval %s(%s, %s)", method, self.expr, other.expr)
             raise
-        out = safe_expand(out)
+        if method not in {"__mul__", "__pow__"}:
+            out = safe_expand(out)
+
         sym_node_log.debug("%s %s %s -> %s", method, self.expr, other.expr, out)
         pytype: Type
         # This is not strictly correct. In Python, a**b may return complex when
