diff --git a/torch/fx/experimental/sym_node.py b/torch/fx/experimental/sym_node.py
index 268f56e3214..95a89347cfe 100644
--- a/torch/fx/experimental/sym_node.py
+++ b/torch/fx/experimental/sym_node.py
@@ -1095,7 +1095,8 @@ def _make_node_magic(method, func):
         except Exception:
             log.warning("failed to eval %s(%s, %s)", method, self.expr, other.expr)
             raise
-        out = safe_expand(out)
+        if method not in {"__mul__", "__pow__"}:
+            out = safe_expand(out)
         sym_node_log.debug("%s %s %s -> %s", method, self.expr, other.expr, out)
         pytype: Type
         # This is not strictly correct. In Python, a**b may return complex when
diff --git a/torch/fx/experimental/symbolic_shapes.py b/torch/fx/experimental/symbolic_shapes.py
index 2b1992fbecb..7b485510874 100644
--- a/torch/fx/experimental/symbolic_shapes.py
+++ b/torch/fx/experimental/symbolic_shapes.py
@@ -1794,7 +1794,7 @@ def _fast_expand(expr: _SympyT) -> _SympyT:
 
 @lru_cache(256)
 def safe_expand(r: _SympyT) -> _SympyT:
-    if hasattr(r, "expand"):
+    if hasattr(r, "expand") and not isinstance(r, (Mul, Pow)):
         try:
             return _fast_expand(r)
         except RecursionError:
