diff --git a/torch/_inductor/codegen/mps.py b/torch/_inductor/codegen/mps.py
index f68c241ca83..291afa5b05b 100644
--- a/torch/_inductor/codegen/mps.py
+++ b/torch/_inductor/codegen/mps.py
@@ -153,6 +153,9 @@ class MetalExprPrinter(ExprPrinter_):
         x = self.doprint(expr.args[0])
         return f"static_cast<int>(metal::trunc({x}))"
 
+    def _print_Float(self, expr: sympy.Expr) -> str:
+        return f"{float(expr)}"
+
     def _print_OpaqueUnaryFn_log2(self, expr: sympy.Expr) -> str:
         assert len(expr.args) == 1
         x = self.doprint(expr.args[0])
diff --git a/torch/_inductor/codegen/triton.py b/torch/_inductor/codegen/triton.py
index c24cde56358..76c5b68d317 100644
--- a/torch/_inductor/codegen/triton.py
+++ b/torch/_inductor/codegen/triton.py
@@ -737,9 +737,9 @@ class TritonPrinter(PythonPrinter):
 
     def _print_Float(self, expr: sympy.Expr) -> str:
         if config.is_fbcode() and torch.version.hip:
-            ret = f"{expr}"
+            ret = f"{float(expr)}"
         else:
-            ret = f"tl.full([], {expr}, tl.float64)"
+            ret = f"tl.full([], {float(expr)}, tl.float64)"
         return ret
 
     def _print_ToFloat(self, expr: sympy.Expr) -> str:
