diff --git a/torch/_dynamo/backends/common.py b/torch/_dynamo/backends/common.py
index 246596bcbca..e26fd5b00b1 100644
--- a/torch/_dynamo/backends/common.py
+++ b/torch/_dynamo/backends/common.py
@@ -69,12 +69,13 @@ class AotAutograd:
         def wrap_bw_compiler(bw_compiler_fn):
             def _wrapped_bw_compiler(*args, **kwargs):
                 # stop TorchDynamo from trying to compile our generated backwards pass
-                return disable(
-                    disable(
-                        bw_compiler_fn, reason="do not trace backward compiler function"
-                    )(*args, **kwargs),
-                    reason="do not trace generated backwards pass",
-                )
+                with torch.no_grad():
+                    return disable(
+                        disable(
+                            bw_compiler_fn, reason="do not trace backward compiler function"
+                        )(*args, **kwargs),
+                        reason="do not trace generated backwards pass",
+                    )
 
             return _wrapped_bw_compiler
 
