diff --git a/torch/_dynamo/variables/functions.py b/torch/_dynamo/variables/functions.py
index 38186fd49f2..e2a71ce309e 100644
--- a/torch/_dynamo/variables/functions.py
+++ b/torch/_dynamo/variables/functions.py
@@ -216,16 +216,18 @@ def bind_args_cached(
             # Maybe override pos-defaults applied above
             ba[name] = wrap_bound_arg(tx, rem_kw.pop(name))
         elif name not in ba:
+            unbound_args = [name]
             raise_observed_exception(
                 TypeError,
                 tx,
                 args=[
                     ConstantVariable.create(
-                        f"Missing required positional argument: {name}"
+                        f"Missing required positional argument: {unbound_args}"
                     )
                 ],
             )
 
+
     # 2) *args
     extra = args[len(spec.all_pos_names) :]
     if spec.varargs_name:
