diff --git a/torch/nested/_internal/ops.py b/torch/nested/_internal/ops.py
index 35e7993454f..3567c2a9308 100644
--- a/torch/nested/_internal/ops.py
+++ b/torch/nested/_internal/ops.py
@@ -2118,6 +2118,8 @@ def all_dim(func, *args, **kwargs):
         torch.ops.aten.any.default,
         torch.ops.aten.max.default,
         torch.ops.aten.min.default,
+        torch.ops.aten._is_any_true,
+        torch.ops.aten._is_all_true,
     ],
     "self: jt_all",
 )
