diff --git a/torch/nested/_internal/ops.py b/torch/nested/_internal/ops.py
index 35e7993454f..af6675a8240 100644
--- a/torch/nested/_internal/ops.py
+++ b/torch/nested/_internal/ops.py
@@ -2116,8 +2116,8 @@ def all_dim(func, *args, **kwargs):
     [
         torch.ops.aten.all.default,
         torch.ops.aten.any.default,
-        torch.ops.aten.max.default,
-        torch.ops.aten.min.default,
+        torch.ops.aten._is_all_true.default,
+        torch.ops.aten._is_any_true.default,
     ],
     "self: jt_all",
 )
