diff --git a/torch/_inductor/codecache.py b/torch/_inductor/codecache.py
index ea69ba454f4..ca83f124538 100644
--- a/torch/_inductor/codecache.py
+++ b/torch/_inductor/codecache.py
@@ -516,7 +516,8 @@ class FxGraphCachePickler(pickle.Pickler):
         only the metadata for these tensors.
         """
         self._stream = io.BytesIO()
-        super().__init__(self._stream)
+        super().__init__(self._stream, protocol=pickle.HIGHEST_PROTOCOL)
+
 
         self.dispatch_table = copyreg.dispatch_table.copy()
         self.dispatch_table.update(
