diff --git a/torch/autograd/__init__.py b/torch/autograd/__init__.py
index c370a0368d7..f311a4bd318 100644
--- a/torch/autograd/__init__.py
+++ b/torch/autograd/__init__.py
@@ -325,8 +325,12 @@ def backward(
                 "arguments both passed to `backward()`. Please only "
                 "use `grad_tensors`."
             )
-    if inputs is not None and len(inputs) == 0:
-        raise RuntimeError("`inputs` argument to `backward()` cannot be empty.")
+    if inputs is not None:
+        if isinstance(inputs, torch.Tensor):
+            inputs = (inputs,)
+        elif len(inputs) == 0:
+            raise RuntimeError("`inputs` argument to `backward()` cannot be empty.")
+
 
     if is_tensor_like(tensors) or isinstance(tensors, graph.GradientEdge):
         tensors = cast(
