diff --git a/torch/_functorch/compile_utils.py b/torch/_functorch/compile_utils.py
index 1ce1f1baf65..6aa0e411719 100644
--- a/torch/_functorch/compile_utils.py
+++ b/torch/_functorch/compile_utils.py
@@ -44,7 +44,7 @@ def fx_graph_cse(fx_g: torch.fx.graph.Graph):
     hash_env = {}  # map from hash to a node in the new graph
     token_map = {}  # map from hash to token
     for n in fx_g.nodes:
-        # The placeholder, output, and get_attr nodes are copied to the new graph without change
+        # The placeholder, output, get_attr nodes, and mutation ops are copied to the new graph without change
         # do not CSE away random operations
         if (
             n.op == "placeholder"
diff --git a/torch/_inductor/pattern_matcher.py b/torch/_inductor/pattern_matcher.py
index 9c825b1d8aa..4de152130fc 100644
--- a/torch/_inductor/pattern_matcher.py
+++ b/torch/_inductor/pattern_matcher.py
@@ -1611,7 +1611,7 @@ def is_mutation_op(node: torch.fx.Node) -> bool:
     elif node.op == "call_method":
         if _mutation_op_re.search(node.target):  # type: ignore[union-attr, arg-type]
             return True
-    return node.kwargs.get("out") is not None
+    return node.kwargs.get("out") is not None or node.op == "call_function" and node.target == torch.ops.fsdp.set_
 
 
 def get_mutation_region_id(graph: torch.fx.Graph, node: torch.fx.Node) -> int:
