diff --git a/torch/_dynamo/symbolic_convert.py b/torch/_dynamo/symbolic_convert.py
index fac68fa5242..7ced3d64b7d 100644
--- a/torch/_dynamo/symbolic_convert.py
+++ b/torch/_dynamo/symbolic_convert.py
@@ -579,7 +579,7 @@ def generic_jump(truth_fn: typing.Callable[[object], bool], push: bool):
                     self.push(value)
                 self.jump(inst)
         elif (
-            isinstance(value, (TensorVariable)) and self.should_compile_partial_graph()
+            isinstance(value, (TensorVariable, ConstDictVariable)) and self.should_compile_partial_graph()
         ):
             jump_graph_break(self, inst, value)
         elif isinstance(value, NNModuleVariable):
