diff --git a/torch/utils/_config_module.py b/torch/utils/_config_module.py
index 41569465d25..90f298f9560 100644
--- a/torch/utils/_config_module.py
+++ b/torch/utils/_config_module.py
@@ -61,7 +61,8 @@ class Config:
         justknob: Optional[str] = None,
         env_name_default: Optional[str] = None,
         env_name_force: Optional[str] = None,
-    ):
+    ) -> None:
+
         # python 3.9 does not support kw_only on the dataclass :(.
         self.default = default
         self.justknob = justknob
