diff --git a/torch/_inductor/standalone_compile.py b/torch/_inductor/standalone_compile.py
index 1ea0f271671..d7a9cc2269c 100644
--- a/torch/_inductor/standalone_compile.py
+++ b/torch/_inductor/standalone_compile.py
@@ -88,8 +88,10 @@ class CompiledArtifact:
                     file.write(writer.to_bytes())
             else:
                 assert format == "unpacked"
-                assert os.path.isdir(path)
-                shutil.rmtree(path, ignore_errors=True)
+                if os.path.exists(path):
+                    assert os.path.isdir(path)
+                    shutil.rmtree(path, ignore_errors=True)
+
 
                 with temporary_cache_dir(path):
                     # This function unpacks the cache artifacts to disk
