diff --git a/torch/_dynamo/variables/builtin.py b/torch/_dynamo/variables/builtin.py
index c6b90d3de2b..7885de4c578 100644
--- a/torch/_dynamo/variables/builtin.py
+++ b/torch/_dynamo/variables/builtin.py
@@ -1415,7 +1415,7 @@ class BuiltinVariable(VariableTracker):
                     out = tx.inline_user_function_return(func_var, args, kwargs)
                     if isinstance(out, ConstDictVariable):
                         return out
-                    return BuiltinVariable(user_cls).call_custom_dict(tx, user_cls, out)
+                    return BuiltinVariable(user_cls).call_custom_dict(tx, user_cls, [out])
         elif not args and kwargs:
             items = {ConstantVariable.create(k): v for k, v in kwargs.items()}
             return variables.ConstDictVariable(
diff --git a/torch/_dynamo/variables/dicts.py b/torch/_dynamo/variables/dicts.py
index a846612e1a5..2d428226413 100644
--- a/torch/_dynamo/variables/dicts.py
+++ b/torch/_dynamo/variables/dicts.py
@@ -88,8 +88,8 @@ class ConstDictVariable(VariableTracker):
             elif isinstance(self.vt, variables.TupleVariable):
                 Hashable = ConstDictVariable._HashableTracker
                 x = tuple(Hashable(e).underlying_value for e in self.vt.items)
-            elif isinstance(self.vt, variables.NNModuleVariable):
-                return self.vt.module
+            elif isinstance(self.vt, variables.NNModuleVariable) or isinstance(self.vt, variables.UnspecializedNNModuleVariable):
+                return self.vt.module if hasattr(self.vt, 'module') else self.vt.value
             elif isinstance(self.vt, variables.UnspecializedNNModuleVariable):
                 return self.vt.value
             elif isinstance(self.vt, variables.UserFunctionVariable):
@@ -295,7 +295,7 @@ class ConstDictVariable(VariableTracker):
             return self.getitem_const_raise_exception_if_absent(tx, args[0])
         elif name == "items":
             assert not (args or kwargs)
-            if self.source:
+            if self.source and isinstance(self.source, ConstDictVariable):
                 tx.output.guard_on_key_order.add(self.source.name())
             return TupleVariable(
                 [TupleVariable([k.vt, v]) for k, v in self.items.items()]
diff --git a/torch/_dynamo/variables/nn_module.py b/torch/_dynamo/variables/nn_module.py
index 5a54031849c..f2b09a81b75 100644
--- a/torch/_dynamo/variables/nn_module.py
+++ b/torch/_dynamo/variables/nn_module.py
@@ -143,7 +143,7 @@ class NNModuleVariable(VariableTracker):
         self.module_type = module_type
         self.module_key = module_key
         self.module = module
-        assert self.source
+        self.nn_module_stack_source = self.source if self.source else None
         self.nn_module_stack_source = self.source
 
     def get_nn_module_stack_source(self):
