diff --git a/torch/_dynamo/variables/builtin.py b/torch/_dynamo/variables/builtin.py
index e927a3df47d..c2e8cd10d0d 100644
--- a/torch/_dynamo/variables/builtin.py
+++ b/torch/_dynamo/variables/builtin.py
@@ -844,6 +844,13 @@ class BuiltinVariable(VariableTracker):
         )
         return abs_method.call_function(tx, [], {})
 
+    def call_pos(self, tx, arg: "VariableTracker"):
+        # Call +arg
+        pos_method = BuiltinVariable(getattr).call_function(
+            tx, [arg, ConstantVariable.create("__pos__")], {}
+        )
+        return pos_method.call_function(tx, [], {})
+
     def call_round(self, tx, arg, *args, **kwargs):
         # Call arg.__round__()
         round_method = BuiltinVariable(getattr).call_function(
@@ -851,6 +858,7 @@ class BuiltinVariable(VariableTracker):
         )
         return round_method.call_function(tx, args, kwargs)
 
+
     def call_range(self, tx, *args):
         if self.unspec_python_args(*args) or self.constant_args(*args):
             return variables.RangeVariable(args)
diff --git a/torch/fx/experimental/sym_node.py b/torch/fx/experimental/sym_node.py
index 3bc3aae1cf1..4ad64f39cee 100644
--- a/torch/fx/experimental/sym_node.py
+++ b/torch/fx/experimental/sym_node.py
@@ -215,9 +215,13 @@ class SymNode:
 
     # These methods call the metaprogrammed methods, they're hand written
     # here so we get good stack traces
+    def pos(self) -> "SymNode":
+        return self  # type: ignore[attr-defined]
+
     def abs(self) -> "SymNode":
         return self._abs()  # type: ignore[attr-defined]
 
+
     def round(self, ndigits=None) -> "SymNode":
         return self._round(ndigits)  # type: ignore[attr-defined]
 
