diff --git a/torch/_higher_order_ops/auto_functionalize.py b/torch/_higher_order_ops/auto_functionalize.py
index 88b32ad70a5..b06bcc6208d 100644
--- a/torch/_higher_order_ops/auto_functionalize.py
+++ b/torch/_higher_order_ops/auto_functionalize.py
@@ -317,6 +317,8 @@ class AutoFunctionalized(HigherOrderOperator):
 
     def __init__(self) -> None:
         super().__init__("auto_functionalized")
+        self._cacheable = True
+
 
     def __call__(
         self,
@@ -346,6 +348,8 @@ class AutoFunctionalizedV2(HigherOrderOperator):
 
     def __init__(self) -> None:
         super().__init__("auto_functionalized_v2")
+        self._cacheable = True
+
 
     def __call__(
         self,
