diff --git a/torch/__init__.py b/torch/__init__.py
index a6c731cd2c1..80534b27f54 100644
--- a/torch/__init__.py
+++ b/torch/__init__.py
@@ -514,10 +514,14 @@ class SymInt:
             # Force specialization
             return hash(builtins.int(self))
 
+    def _sympy_(self):
+        return self.node.expr
+
 
 class SymFloat:
     """
     Like an float (including magic methods), but redirects all operations on the
+
     wrapped node. This is used in particular to symbolically record operations
     in the symbolic shape workflow.
     """
@@ -622,10 +626,14 @@ class SymFloat:
             # Force specialization
             return hash(builtins.float(self))
 
+    def _sympy_(self):
+        return self.node.expr
+
 
 class SymBool:
     """
     Like an bool (including magic methods), but redirects all operations on the
+
     wrapped node. This is used in particular to symbolically record operations
     in the symbolic shape workflow.
 
