diff --git a/torch/package/package_importer.py b/torch/package/package_importer.py
index 72bde854cba..18be579c6b8 100644
--- a/torch/package/package_importer.py
+++ b/torch/package/package_importer.py
@@ -243,14 +243,19 @@ class PackageImporter(Importer):
             name = f"{key}.storage"
 
             if storage_context.has_storage(name):
-                storage = storage_context.get_storage(name, dtype)._typed_storage()
+                storage = storage_context.get_storage(name, dtype)
+                if dtype is not None:
+                    storage = storage._typed_storage()
             else:
                 tensor = self.zip_reader.get_storage_from_record(
                     ".data/" + name, size, dtype
                 )
                 if isinstance(self.zip_reader, torch._C.PyTorchFileReader):
                     storage_context.add_storage(name, tensor)
-                storage = tensor._typed_storage()
+                if dtype is not None:
+                    storage = tensor._typed_storage()
+                else:
+                    storage = tensor
             loaded_storages[key] = restore_location(storage, location)
 
         def persistent_load(saved_id):
