diff --git a/torch/_inductor/codegen/cpp.py b/torch/_inductor/codegen/cpp.py
index f5692a838ec..51e61e6815e 100644
--- a/torch/_inductor/codegen/cpp.py
+++ b/torch/_inductor/codegen/cpp.py
@@ -599,6 +599,8 @@ class CppOverrides(OpOverrides):
 
     @staticmethod
     def add(a, b):
+        if isinstance(a, int) and isinstance(b, int) and (b < 0 or b >= 64):
+            return f"decltype({a})(0)"
         return f"decltype({a})({a} + {b})"
 
     @staticmethod
