diff --git a/torch/ao/quantization/fx/_lower_to_native_backend.py b/torch/ao/quantization/fx/_lower_to_native_backend.py
index aa71fafbf00..cf770eef57f 100644
--- a/torch/ao/quantization/fx/_lower_to_native_backend.py
+++ b/torch/ao/quantization/fx/_lower_to_native_backend.py
@@ -110,6 +110,7 @@ def is_copy_node(node, modules):
         torch.clamp,
         torch.flatten,
         torch.mean,
+        torch.nn.functional.channel_shuffle,
         operator.floordiv,
     ]
     method_list = [
@@ -134,6 +135,7 @@ def is_copy_node(node, modules):
     ]
     return _is_node_in_list(node, modules, func_list, method_list, module_type_list)
 
+
 def is_general_tensor_shape_node(node, modules):
     func_list = [
         torch.transpose,
