diff --git a/torch/_inductor/lowering.py b/torch/_inductor/lowering.py
index 97aa63705f1..ff365622dd5 100644
--- a/torch/_inductor/lowering.py
+++ b/torch/_inductor/lowering.py
@@ -1906,7 +1906,7 @@ def full(size, fill_value, **kwargs):
 
 
 @register_lowering(aten.gather, type_promotion_kind=None)
-def gather(x, dim, index):
+def gather(x, dim, index, *, sparse_grad=False):
     assert isinstance(x, TensorBox)
     assert index.get_dtype() == torch.int64
     offset = len(x.get_size()) == 0
