diff --git a/torch/_inductor/lowering.py b/torch/_inductor/lowering.py
index 3ab0a30a402..465ce2cf95d 100644
--- a/torch/_inductor/lowering.py
+++ b/torch/_inductor/lowering.py
@@ -3061,7 +3061,8 @@ make_fallback(aten._efficient_attention_backward.default, sdpa_constraint)
 
 # index_reduce requires fallback when use_scatter_fallback(...) returns True
 make_fallback(aten.index_reduce)
-make_fallback(aten.repeat_interleave.Tensor, override_decomp=True)
+make_fallback(aten.repeat_interleave.Tensor, override_decomp)
+make_fallback(aten.copy_, override_decomp)=True)
 
 
 # Register with type_promotion_kind None.
