diff --git a/torch/_inductor/autotune_process.py b/torch/_inductor/autotune_process.py
index 1a417117f4a..bd959dd2be8 100644
--- a/torch/_inductor/autotune_process.py
+++ b/torch/_inductor/autotune_process.py
@@ -585,7 +585,8 @@ class TritonBenchmarkRequest(BenchmarkRequest):
         kpack: int = 0,  # ROCm specific gemm paramete
         workspace_arg: Optional[WorkspaceArg] = None,
     ) -> None:
-        super().__init__(kernel_name, input_tensor_meta, output_tensor_meta, extra_args)
+        super().__init__(kernel_name, input_tensor_meta, output_tensor_meta, extra_args, workspace_arg)
+
         self.module_path = module_path
         self.module_cache_key = module_cache_key
         self.num_stages = num_stages
