diff --git a/torch/export/exported_program.py b/torch/export/exported_program.py
index 02c00002f5d..181464484fa 100644
--- a/torch/export/exported_program.py
+++ b/torch/export/exported_program.py
@@ -594,12 +594,9 @@ def _decompose_and_get_gm_with_new_signature_constants(
 
     output_specs = [
         OutputSpec(
-            OutputKind.LOSS_OUTPUT if joint_loss_index is not None else spec.kind,
-            update_arg(spec.arg, new_outputs[i]),
-            old_new_placeholder_map.get(spec.target, spec.target),
-        )
-        for i, spec in enumerate(ep.graph_signature.output_specs)
-    ]
+            OutputKind.LOSS_OUTPUT if (joint_loss_index is not None and i == joint_loss_index) else spec.kind,
+            update_arg(spec.arg, new_output
+
 
     if joint_loss_index is not None:
         assert graph_signature.backward_signature is not None
