diff --git a/torch/_refs/special/__init__.py b/torch/_refs/special/__init__.py
index 3599f2fd85c..b9e623bc308 100644
--- a/torch/_refs/special/__init__.py
+++ b/torch/_refs/special/__init__.py
@@ -133,11 +133,10 @@ def log_ndtr(a: TensorLikeType) -> TensorLikeType:
     type_promotion_kind=utils.ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT,
 )
 def logit(self: TensorLikeType, eps: Optional[float] = None) -> TensorLikeType:
-    if eps is None:
-        eps = -1.0
-    lo = eps
-    hi = 1 - eps
-    self = torch.clamp(self, lo, hi)
+    if eps is not None:
+        lo = eps
+        hi = 1 - eps
+        self = torch.clamp(self, lo, hi)
     return torch.log(torch.true_divide(self, torch.sub(1, self)))
 
 
