diff --git a/torch/_dynamo/variables/builder.py b/torch/_dynamo/variables/builder.py
index 3dd41a1d91f..07186612c91 100644
--- a/torch/_dynamo/variables/builder.py
+++ b/torch/_dynamo/variables/builder.py
@@ -576,7 +576,8 @@ class VariableBuilder:
         ):
             return self.wrap_tensor(value)
         elif is_namedtuple(value):
-            return self.wrap_listlike(value)
+            return self.wrap_listlike(value, source=self.source)
+
 
         elif value is torch.utils._pytree.SUPPORTED_NODES:
             # For SUPPORTED_NODES, we guard on the dictionary version (PEP509)
