diff --git a/torch/serialization.py b/torch/serialization.py
index 8d8ae774e3d..f2c36d847ef 100644
--- a/torch/serialization.py
+++ b/torch/serialization.py
@@ -342,7 +342,9 @@ def get_unsafe_globals_in_checkpoint(f: FILE_LIKE) -> List[str]:
     Returns:
         A list of strings of pickle GLOBALs in the checkpoint that are not allowlisted for ``weights_only``.
     """
-    safe_global_strings = set(_weights_only_unpickler._get_allowed_globals().keys())
+    safe_global_strings = set(_weights_only_unpickler._get_allowed_globals().keys()).union(
+        {f"{g.__module__}.{g.__name__}" for g in get_safe_globals()}
+    )
 
     with _open_file_like(f, "rb") as opened_file:
         if not _is_zipfile(opened_file):
