diff --git a/torch/multiprocessing/reductions.py b/torch/multiprocessing/reductions.py
index 727b4f3fc37..19e2768a4dc 100644
--- a/torch/multiprocessing/reductions.py
+++ b/torch/multiprocessing/reductions.py
@@ -109,7 +109,10 @@ def reduce_event(event):
 
 def rebuild_tensor(cls, storage, metadata):
     storage_offset, size, stride, requires_grad = metadata
-    t = torch._utils._rebuild_tensor(storage, storage_offset, size, stride)
+    if storage.device.type == 'meta':
+        t = torch.empty_strided(size, stride, dtype=storage.dtype, device='meta')
+    else:
+        t = torch._utils._rebuild_tensor(storage, storage_offset, size, stride)
     if cls == torch.nn.parameter.Parameter:
         # we have to pass requires_grad into constructor, rather than set it as an
         # attribute later, because it's an important check for Integer Tensors to
