diff --git a/torch/_inductor/pattern_matcher.py b/torch/_inductor/pattern_matcher.py
index db96c45c4a8..a546e7b91a1 100644
--- a/torch/_inductor/pattern_matcher.py
+++ b/torch/_inductor/pattern_matcher.py
@@ -202,7 +202,7 @@ class Match:
 
     def erase_nodes(self, graph: torch.fx.Graph) -> None:
         for n in reversed(self.nodes):
-            if not n._erased:
+            if not n._erased and len(n.users) == 0:
                 graph.erase_node(n)
 
     def output_nodes(self) -> List[Optional[torch.fx.Node]]:
