diff --git a/torch/_library/utils.py b/torch/_library/utils.py
index 7b8eec899d7..ca9a75c00cd 100644
--- a/torch/_library/utils.py
+++ b/torch/_library/utils.py
@@ -188,9 +188,10 @@ def zip_schema(
     assert len(schema.arguments) >= len(args) + len(kwargs)
     for i in range(len(schema.arguments)):
         info = schema.arguments[i]
+        if info.name in kwargs:
+            yield info, kwargs[info.name]
+            continue
         if info.kwarg_only:
-            if info.name in kwargs:
-                yield info, kwargs[info.name]
             continue
         if i >= len(args):
             # args that are equal to their default values are not populated
@@ -201,6 +202,7 @@ def zip_schema(
     return
 
 
+
 def hop_schema_from_fx_node(node):
     from torchgen.gen_schema_utils import FunctionSchemaGen
 
