diff --git a/torch/_dynamo/variables/tensor.py b/torch/_dynamo/variables/tensor.py
index 3da23d7731e..a555b1f82c6 100644
--- a/torch/_dynamo/variables/tensor.py
+++ b/torch/_dynamo/variables/tensor.py
@@ -69,6 +69,8 @@ supported_tensor_comparison_ops = {
     "<=": operator.le,
     "==": operator.eq,
     "!=": operator.ne,
+    "is": operator.is_,
+    "is not": operator.is_not,
 }
 # Ops that allow tensor <op> None
 supported_const_comparison_ops = {
