diff --git a/torch/_dynamo/symbolic_convert.py b/torch/_dynamo/symbolic_convert.py
index 2c2e824664a..136fd73537a 100644
--- a/torch/_dynamo/symbolic_convert.py
+++ b/torch/_dynamo/symbolic_convert.py
@@ -1938,7 +1938,7 @@ class InstructionTranslatorBase(
 
         value = self.pop()
         if isinstance(value, SymNodeVariable):
-            value = ConstantVariable.create(str(value.sym_num))
+            value = LazySymNodeFormatString(value)
         if (flags & 0x03) == 0x01:
             value = BuiltinVariable(str).call_function(self, [value], {})  # type: ignore[arg-type]
         elif (flags & 0x03) == 0x02:
diff --git a/torch/_dynamo/variables/lazy.py b/torch/_dynamo/variables/lazy.py
index 04be291cf5c..70349a63936 100644
--- a/torch/_dynamo/variables/lazy.py
+++ b/torch/_dynamo/variables/lazy.py
@@ -27,6 +27,13 @@ class LazyCache:
         del self.source
 
 
+class LazySymNodeFormatString:
+    def __init__(self, sym_node):
+        self.sym_node = sym_node
+
+    def __str__(self):
+        return str(self.sym_node.sym_num)
+
 class LazyVariableTracker(VariableTracker):
     """
     A structure that defers the creation of the actual VariableTracker
