diff --git a/torch/_dynamo/trace_rules.py b/torch/_dynamo/trace_rules.py
index 6be6e4965ce..017087a17e3 100644
--- a/torch/_dynamo/trace_rules.py
+++ b/torch/_dynamo/trace_rules.py
@@ -3091,7 +3091,7 @@ def is_builtin_constant(obj) -> bool:
 def is_numpy(obj) -> bool:
     if np is None:
         return False
-    return isinstance(obj, (np.ndarray, np.generic)) or id(obj) in _numpy_function_ids
+    return isinstance(obj, (np.ndarray, np.generic, np.dtype, np.iinfo, np.finfo)) or id(obj) in _numpy_function_ids
 
 
 BUILTIN_SKIPLIST = (
diff --git a/torch/_dynamo/variables/builder.py b/torch/_dynamo/variables/builder.py
index 10a79ed8ff3..d79c4665b14 100644
--- a/torch/_dynamo/variables/builder.py
+++ b/torch/_dynamo/variables/builder.py
@@ -70,7 +70,7 @@ from ..source import (
     Source,
     TupleIteratorGetItemSource,
 )
-from ..trace_rules import is_callable_allowed, is_numpy
+from ..trace_rules import is_callable_allowed, is_numpy, is_numpy_dtype, is_numpy_type_info
 from ..utils import (
     build_checkpoint_variable,
     clone_input,
@@ -160,7 +160,7 @@ from .misc import (
 )
 from .nn_module import FSDPManagedNNModuleVariable, UnspecializedNNModuleVariable
 from .optimizer import OptimizerVariable
-from .script_object import TorchScriptObjectVariable
+from .script_object import TorchScriptObjectVariable, NumpyDtypeVariable, NumpyTypeInfoVariable
 
 from .sdpa import SDPAParamsVariable
 from .tensor import (
