diff --git a/torch/_inductor/ir.py b/torch/_inductor/ir.py
index 52cb8af69fd..92c1315fa74 100644
--- a/torch/_inductor/ir.py
+++ b/torch/_inductor/ir.py
@@ -7010,6 +7010,7 @@ class DeviceCopy(ExternKernelOut):
                 device=device,
                 dtype=x.get_dtype(),
                 size=x.get_size(),
+                stride=x.get_stride(),
             ),
             [cls.realize_input(x)],
             constant_args,
diff --git a/torch/_meta_registrations.py b/torch/_meta_registrations.py
index e33b87895f3..459d30320a4 100644
--- a/torch/_meta_registrations.py
+++ b/torch/_meta_registrations.py
@@ -7025,8 +7025,7 @@ def _amp_foreach_non_finite_check_and_unscale_(self, found_inf, inv_scale):
 @register_meta([aten.nan_to_num.default, aten.nan_to_num.out])
 @out_wrapper()
 def nan_to_num(self, nan=None, posinf=None, neginf=None):
-    result_size = list(self.size())
-    return self.new_empty(result_size)
+    return self.new_empty_strided(self.size(), self.stride())
 
 
 @register_meta(torch.ops.aten.transpose_)
