diff --git a/torch/_inductor/ir.py b/torch/_inductor/ir.py
index 52cb8af69fd..b217910f3b6 100644
--- a/torch/_inductor/ir.py
+++ b/torch/_inductor/ir.py
@@ -7006,7 +7006,7 @@ class DeviceCopy(ExternKernelOut):
         developer_warning("DeviceCopy in input program")
         constant_args = (non_blocking,)
         return DeviceCopy(
-            FlexibleLayout(
+            x.get_layout().make_strided(),
                 device=device,
                 dtype=x.get_dtype(),
                 size=x.get_size(),
diff --git a/torch/_meta_registrations.py b/torch/_meta_registrations.py
index e33b87895f3..03c01ecb178 100644
--- a/torch/_meta_registrations.py
+++ b/torch/_meta_registrations.py
@@ -705,7 +705,7 @@ def meta_index_select(self, dim, index):
     result_size = list(self.size())
     if self.dim() > 0:
         result_size[dim] = index.numel()
-    return self.new_empty(result_size)
+    return self.new_empty(result_size, layout=self.layout)
 
 
 @register_meta(aten.segment_reduce.default)
