diff --git a/torch/nested/_internal/ops.py b/torch/nested/_internal/ops.py
index dab2ec46fb7..a175b010ad0 100644
--- a/torch/nested/_internal/ops.py
+++ b/torch/nested/_internal/ops.py
@@ -224,7 +224,9 @@ def lookup_jagged(func, *args, **kwargs) -> Optional[Callable]:
             check_schema(check_schema_str, func, *args, **kwargs)
             return functools.partial(jagged_unary_pointwise, func)
         elif num_tensor_args == 2:
-            check_schema("lhs: any, rhs: any, ...", func, *args, **kwargs)
+            if not all(isinstance(arg, torch.Tensor) for arg in args[:2]):
+                raise RuntimeError("invalid argument")
+            check_schema("lhs: jt_all, rhs: jt_all, ...", func, *args, **kwargs)
             return functools.partial(jagged_binary_pointwise, func)
 
     return None
