diff --git a/torch/optim/lr_scheduler.py b/torch/optim/lr_scheduler.py
index 06ae24ef08d..7aed30f37f2 100644
--- a/torch/optim/lr_scheduler.py
+++ b/torch/optim/lr_scheduler.py
@@ -300,8 +300,14 @@ class LambdaLR(LRScheduler):
                     f"Expected {len(optimizer.param_groups)} lr_lambdas, but got {len(lr_lambda)}"
                 )
             self.lr_lambdas = list(lr_lambda)
+
+        if not all(callable(lmbda) for lmbda in self.lr_lambdas):
+            raise TypeError(
+                f"lr_lambda must be callable or a list of callables, but got {lr_lambda}"
+            )
         super().__init__(optimizer, last_epoch)
 
+
     @override
     def state_dict(self) -> dict[str, Any]:
         """Return the state of the scheduler as a :class:`dict`.
