diff --git a/torch/export/pt2_archive/_package.py b/torch/export/pt2_archive/_package.py
index 84dc4784fa9..cf032ee5f14 100644
--- a/torch/export/pt2_archive/_package.py
+++ b/torch/export/pt2_archive/_package.py
@@ -794,7 +794,7 @@ def _load_state_dict(
                     ),
                 )
                 if payload_meta.is_param:
-                    state_dict[weight_fqn] = torch.nn.Parameter(weight_tensor)
+                    state_dict[weight_fqn] = torch.nn.Parameter(weight_tensor, requires_grad=weight_tensor.requires_grad)
                 else:
                     state_dict[weight_fqn] = weight_tensor
 
