diff --git a/torch/_functorch/compile_utils.py b/torch/_functorch/compile_utils.py
index a5c3201c4ce..65880606389 100644
--- a/torch/_functorch/compile_utils.py
+++ b/torch/_functorch/compile_utils.py
@@ -54,7 +54,7 @@ def fx_graph_cse(fx_g: torch.fx.graph.Graph):
                      "kwargs": kwargs, "kwargs_spec": kwargs_spec}
 
             # hash substituted args to a number, do not hash specs because specs are not hashable
-            hash_arg = hash((args, kwargs))
+            hash_arg = hash((args, kwargs, args_spec, kwargs_spec))
             hash_val = (n.target, hash_arg)
 
             # check if a node has a substitute and can be eliminated
