diff --git a/torch/_dynamo/variables/dicts.py b/torch/_dynamo/variables/dicts.py
index ce60951c2f9..3911f74d0f1 100644
--- a/torch/_dynamo/variables/dicts.py
+++ b/torch/_dynamo/variables/dicts.py
@@ -536,24 +536,14 @@ class SetVariable(ConstDictVariable):
             return variables.UserFunctionVariable(
                 polyfills.set_difference
             ).call_function(tx, [self, args[0]], {})
-        elif (
-            name == "update"
-            and len(args) == 1
-            and isinstance(
-                args[0],
-                (
-                    SetVariable,
-                    ListVariable,
-                    TupleVariable,
-                ),
+        elif name == "update" and self.is_mutable():
+            assert not kwargs
+            assert len(args) == 1
+            union = variables.UserFunctionVariable(polyfills.set_union).call_function(
+                tx, [self, args[0]], {}
             )
-            and self.is_mutable()
-        ):
-            if isinstance(args[0], (ListVariable, TupleVariable)):
-                arg = SetVariable(args[0].unpack_var_sequence(tx))
-            else:
-                arg = args[0]
-            return super().call_method(tx, "update", (arg,), kwargs)
+            self.set_items = union.set_items
+            return self
         elif name == "remove":
             assert not kwargs
             assert len(args) == 1
