diff --git a/torch/_dynamo/guards.py b/torch/_dynamo/guards.py
index 8c20aa33810..e87432f4998 100644
--- a/torch/_dynamo/guards.py
+++ b/torch/_dynamo/guards.py
@@ -1794,11 +1794,6 @@ class GuardBuilder(GuardBuilderBase):
         )
 
     def TENSOR_MATCH(self, guard: Guard, value=None):
-        # For FSDP modules, we can skip guards on nn module tensors because FSDP
-        # eager assumes that the params are unchanged once the model is wrapped.
-        if guard.is_fsdp_module():
-            return
-
         # For tensors that are part of the Dynamo extracted Fx graph module, an
         # ID_MATCH suffices. Once we turn on inline_inbuilt_nn_modules, these
         # will be lifted as inputs and have a TENSOR_MATCH guard.
diff --git a/torch/_guards.py b/torch/_guards.py
index c42d4837500..b4bac3c77b1 100644
--- a/torch/_guards.py
+++ b/torch/_guards.py
@@ -100,14 +100,9 @@ class GuardSource(enum.Enum):
         return self in (GuardSource.GLOBAL_FSDP_MODULE, GuardSource.LOCAL_FSDP_MODULE)
 
     def is_specialized_nn_module(self) -> bool:
-        return (
-            self
-            in (
-                GuardSource.GLOBAL_SPECIALIZED_NN_MODULE,
-                GuardSource.LOCAL_SPECIALIZED_NN_MODULE,
-            )
-            # TODO (anijain2305) - Investigate why is_fsdp_module required.
-            or self.is_fsdp_module()
+        return self in (
+            GuardSource.GLOBAL_SPECIALIZED_NN_MODULE,
+            GuardSource.LOCAL_SPECIALIZED_NN_MODULE,
         )
 
     def is_unspecialized_nn_module(self) -> bool:
