diff --git a/torch/_meta_registrations.py b/torch/_meta_registrations.py
index 2f1a88461e4..8bce82e31df 100644
--- a/torch/_meta_registrations.py
+++ b/torch/_meta_registrations.py
@@ -7803,6 +7803,10 @@ _create_unary_float_meta_func(aten.special_scaled_modified_bessel_k1)
 
 
 _create_binary_float_meta_func(aten.special_chebyshev_polynomial_t)
+
+@register_meta(aten._native_multi_head_attention.default)
+def meta__native_multi_head_attention(query, key, value, embed_dim, num_heads, attn_mask=None, bias=None, need_weights=True, average_attn_weights=True, is_causal=False, dropout_p=0.0, training=True, scale=None):
+    #
 _create_binary_float_meta_func(aten.special_chebyshev_polynomial_u)
 _create_binary_float_meta_func(aten.special_chebyshev_polynomial_v)
 _create_binary_float_meta_func(aten.special_chebyshev_polynomial_w)
