diff --git a/torch/_dynamo/variables/tensor.py b/torch/_dynamo/variables/tensor.py
index c782a19d7f1..1631fc71233 100644
--- a/torch/_dynamo/variables/tensor.py
+++ b/torch/_dynamo/variables/tensor.py
@@ -1149,7 +1149,7 @@ class SymNodeVariable(VariableTracker):
     def as_tensor(self, tx):
         if self._tensor_var is None:
             self._tensor_var = VariableTracker.build(
-                tx, torch.scalar_tensor
+                tx, lambda x: torch.scalar_tensor(x, dtype=torch.float64)
             ).call_function(tx, [self], {})
         return self._tensor_var
 
