diff --git a/torch/_weights_only_unpickler.py b/torch/_weights_only_unpickler.py
index 3307052c91d..7719b7af233 100644
--- a/torch/_weights_only_unpickler.py
+++ b/torch/_weights_only_unpickler.py
@@ -141,7 +141,7 @@ def _get_user_allowed_globals():
             f, name = f
             rc[name] = f
         else:
-            module, name = f.__module__, f.__name__
+            module, name = f.__module__, f.__qualname__
             rc[f"{module}.{name}"] = f
     return rc
 
@@ -362,7 +362,7 @@ class Unpickler:
                     )
                 else:
                     raise UnpicklingError(
-                        f"Unsupported global: GLOBAL {full_path} was not an allowed global by default. "
+                        f"Unsupported global: GLOBAL {name} was not an allowed global by default. "
                         f"Please use `torch.serialization.add_safe_globals([{name}])` or the "
                         f"`torch.serialization.safe_globals([{name}])` context manager to allowlist this global "
                         "if you trust this class/function."
