diff --git a/torch/_subclasses/fake_impls.py b/torch/_subclasses/fake_impls.py
index 0b000cfa1a9..768c5ca5440 100644
--- a/torch/_subclasses/fake_impls.py
+++ b/torch/_subclasses/fake_impls.py
@@ -1344,12 +1344,12 @@ def make_fast_binary_impl(
             elif common_dtype != op.dtype:
                 has_different_input_dtypes = True
 
-        if has_different_input_dtypes:
-            # compute promotion
-            # TODO: we don't need the compute type
-            _, common_dtype = elementwise_dtypes(
-                *operands, type_promotion_kind=type_promotion_kind
-            )
+        # compute promotion
+        # TODO: we don't need the compute type
+        _, common_dtype = elementwise_dtypes(
+            *operands, type_promotion_kind=type_promotion_kind
+        )
+
 
         # check all tensors on same device
         # cpu scalars are assumed allow
