diff --git a/torch/_inductor/codegen/cpp.py b/torch/_inductor/codegen/cpp.py
index 431087fe785..8f530508ce8 100644
--- a/torch/_inductor/codegen/cpp.py
+++ b/torch/_inductor/codegen/cpp.py
@@ -820,7 +820,8 @@ class CppOverrides(OpOverrides):
 
     @staticmethod
     def asinh(x):
-        return f"std::asinh({x})"
+        return f"Sleef_asinhf_u10({x})"
+
 
     @staticmethod
     def atan2(x, y):
