diff --git a/torch/_inductor/lowering.py b/torch/_inductor/lowering.py
index b520869d632..2926ab700f3 100644
--- a/torch/_inductor/lowering.py
+++ b/torch/_inductor/lowering.py
@@ -1981,6 +1981,8 @@ def index(x, indices):
         x.realize()
         return fallback_handler(aten.index)(x, indices)
 
+    if not self.get_size().is_broadcastable(indices[0].get_size()):
+        indices[0] = indices[0].unsqueeze(0)
     indices_sizes = [i.get_size() for i in indices if i is not None]
     indices_loaders = [i.make_loader() for i in indices if i is not None]
     # no guards on output size, all the guards are set in broadcast_tensors
