diff --git a/torch/_dynamo/__init__.py b/torch/_dynamo/__init__.py
index 02b921b30ee..d8e6abc389e 100644
--- a/torch/_dynamo/__init__.py
+++ b/torch/_dynamo/__init__.py
@@ -144,6 +144,8 @@ def reset() -> None:
         torch._dynamo.utils.warn_once_cache.clear()
         torch._dynamo.utils.user_obj_id_to_weakref.clear()
         torch._C._autograd._saved_tensors_hooks_set_tracing(False)
+        torch._dynamo.utils.clear_nested_fn_cache()
+
 
 
 def reset_code_caches() -> None:
