diff --git a/torch/_dynamo/__init__.py b/torch/_dynamo/__init__.py
index 02b921b30ee..c75c225c09d 100644
--- a/torch/_dynamo/__init__.py
+++ b/torch/_dynamo/__init__.py
@@ -51,6 +51,7 @@ from .mutation_guard import GenerationTracker
 from .pgo import reset_code_state
 from .symbolic_convert import TensorifyState
 from .utils import (
+    CreateNestedFnCache,
     graph_break_reasons,
     guard_failures,
     orig_code_map,
@@ -143,7 +144,7 @@ def reset() -> None:
         TensorifyState.clear()
         torch._dynamo.utils.warn_once_cache.clear()
         torch._dynamo.utils.user_obj_id_to_weakref.clear()
-        torch._C._autograd._saved_tensors_hooks_set_tracing(False)
+        CreateNestedFnCache.clear()
 
 
 def reset_code_caches() -> None:
diff --git a/torch/_dynamo/utils.py b/torch/_dynamo/utils.py
index 1775dad56d0..0db3c6d8662 100644
--- a/torch/_dynamo/utils.py
+++ b/torch/_dynamo/utils.py
@@ -4825,14 +4825,17 @@ def record_pregraph_bytecode_enter() -> AbstractContextManager[None]:
     return cm
 
 
-@torch._disable_dynamo
-def record_pregraph_bytecode_exit(cm: AbstractContextManager[None]) -> None:
-    cm.__exit__(None, None, None)
+class CreateNestedFnCache:
+    _cache = {}
 
+    @classmethod
+    def clear(cls):
+        cls._cache.clear()
 
-# Returns a set of code objects present traced in the current TracingContext, or None
-# if there is no current TracingContext.
-def get_traced_code() -> Optional[list[CodeType]]:
-    from torch._guards import TracingContext
+    @classmethod
+    def get(cls, key):
+        return cls._cache.get(key)
 
-    return TracingContext.get_traced_code()
+    @classmethod
+    def set(cls, key, value):
+        cls._cache[key] = value
diff --git a/torch/_dynamo/variables/functions.py b/torch/_dynamo/variables/functions.py
index 6eb7d0666cd..b0feccb2b1c 100644
--- a/torch/_dynamo/variables/functions.py
+++ b/torch/_dynamo/variables/functions.py
@@ -58,7 +58,7 @@ from ..exc import (
 from ..guards import GuardBuilder, install_guard
 from ..source import (
     AttrSource,
-    ClosureSource,
+    CreateNestedFnCache,
     ConstantSource,
     DefaultsSource,
     GetItemSource,
@@ -276,7 +276,13 @@ def _create_nested_fn(
 ):
     from types import FunctionType
 
+    cache_key = (code, f_globals, name, defaults, closure, kwdefaults, annotations)
+    cached_func = CreateNestedFnCache.get(cache_key)
+    if cached_func is not None:
+        return cached_func
+
     func = FunctionType(code, f_globals, name, defaults, closure)
+    CreateNestedFnCache.set(cache_key, func)
     func.__kwdefaults__ = kwdefaults
 
     if isinstance(annotations, tuple):
