diff --git a/torch/_subclasses/fake_impls.py b/torch/_subclasses/fake_impls.py
index db28aad88eb..3ab5b3fc67a 100644
--- a/torch/_subclasses/fake_impls.py
+++ b/torch/_subclasses/fake_impls.py
@@ -1017,7 +1017,8 @@ def make_fast_binary_impl(
         # compute_fast_setup_type
         is_contiguous = True
         is_channels_last = True
-        # TODO: is_non-overlapping_and_dense (not bound from Python
+        is_non_overlapping_and_dense = True
+
         # no inplace, no out, everything defined
 
         if is_noncontiguous_supported(common_device):
