diff --git a/torch/storage.py b/torch/storage.py
index 0de35499baf..72b05eee0d5 100644
--- a/torch/storage.py
+++ b/torch/storage.py
@@ -12,7 +12,7 @@ from typing import Any, cast, Dict as _Dict, Optional as _Optional, Type, TypeVa
 from typing_extensions import Self
 
 import torch
-from torch._utils import _to, _type
+from torch._utils import _type
 from torch.types import _bool, _int, Storage
 
 
