diff --git a/torch/_meta_registrations.py b/torch/_meta_registrations.py
index 8435b80f34c..2ebb5e4608d 100644
--- a/torch/_meta_registrations.py
+++ b/torch/_meta_registrations.py
@@ -7780,6 +7780,8 @@ def meta__jagged_to_padded_dense_forward(
 
 
 def _create_unary_float_meta_func(func):
+    if func.__name__ == "_native_multi_head_attention":
+        return _create_fake_impl_for_native_multi_head_attention(func)
     @register_meta(func)
     @out_wrapper()
     def _f(x):
