diff --git a/torch/_higher_order_ops/cond.py b/torch/_higher_order_ops/cond.py
index f0a260de406..bd4ad9784b7 100644
--- a/torch/_higher_order_ops/cond.py
+++ b/torch/_higher_order_ops/cond.py
@@ -62,7 +62,7 @@ cond_op = CondOp()
 
 
 @exposed_in("torch")
-def cond(pred, true_fn, false_fn, operands):
+def cond(pred, true_fn, false_fn, operands=()):
     r"""
     Conditionally applies `true_fn` or `false_fn`.
 
