diff --git a/torch/_higher_order_ops/cond.py b/torch/_higher_order_ops/cond.py
index f0a260de406..64868ead38e 100644
--- a/torch/_higher_order_ops/cond.py
+++ b/torch/_higher_order_ops/cond.py
@@ -54,7 +54,7 @@ class CondOp(HigherOrderOperator):
     def __init__(self):
         super().__init__("cond")
 
-    def __call__(self, pred, true_fn, false_fn, operands):
+    def __call__(self, pred, true_fn, false_fn, operands=()):
         return super().__call__(pred, true_fn, false_fn, operands)
 
 
@@ -62,7 +62,7 @@ cond_op = CondOp()
 
 
 @exposed_in("torch")
-def cond(pred, true_fn, false_fn, operands):
+def cond(pred, true_fn, false_fn, operands=()):
     r"""
     Conditionally applies `true_fn` or `false_fn`.
 
diff --git a/torch/_higher_order_ops/while_loop.py b/torch/_higher_order_ops/while_loop.py
index f14321842f4..6797999b727 100644
--- a/torch/_higher_order_ops/while_loop.py
+++ b/torch/_higher_order_ops/while_loop.py
@@ -44,7 +44,7 @@ class WhileLoopOp(HigherOrderOperator):
             )
         if not all(
             isinstance(t, (torch.Tensor, int, float, bool)) for t in carried_inputs
-        ):
+        )):
             raise RuntimeError(
                 "carried_inputs must be a tuple of tensors, ints, floats, or bools, got "
                 f"{carried_inputs}"
@@ -131,7 +131,7 @@ def while_loop(cond_fn, body_fn, carried_inputs):
         if not callable(cond_fn) or not callable(body_fn):
             raise RuntimeError("Expect cond_fn and body_fn to be callbale.")
 
-        if not isinstance(carried_inputs, (tuple, list)) or pytree.tree_any(
+        if carried_inputs is not None and (not isinstance(carried_inputs, (tuple, list)) or pytree.tree_any(
             lambda t: not isinstance(t, torch.Tensor), carried_inputs
         ):
             raise RuntimeError(
