diff --git a/torch/_inductor/index_propagation.py b/torch/_inductor/index_propagation.py
index 16d5770a36a..eea68951357 100644
--- a/torch/_inductor/index_propagation.py
+++ b/torch/_inductor/index_propagation.py
@@ -324,13 +324,17 @@ class IndexPropagation:
                 for k, v in self.indirect_var_ranges.items()
             ),
         )
-        evaluated = self.shape_env._maybe_evaluate_static(
-            e,
-            axioms=self.axioms,
-            var_to_range=var_to_range,
-        )
+        try:
+            evaluated = self.shape_env._maybe_evaluate_static(
+                e,
+                axioms=self.axioms,
+                var_to_range=var_to_range,
+            )
+        except Exception:
+            return False
         return bool(evaluated)
 
+
     def indirect_indexing(
         self,
         index: Union[Any, IndexPropVar],
