diff --git a/torch/_dynamo/side_effects.py b/torch/_dynamo/side_effects.py
index 2cf79ebbd66..6bf21442013 100644
--- a/torch/_dynamo/side_effects.py
+++ b/torch/_dynamo/side_effects.py
@@ -749,7 +749,8 @@ class SideEffects:
                         cg(var.source)
                         suffixes.append([create_instruction("STORE_ATTR", argval=name)])
             elif isinstance(var, variables.TupleIteratorVariable):
-                for _ in range(var.index):
+                for _ in range(var.index + 1):
+
                     cg.add_push_null(
                         lambda: cg.load_import_from(utils.__name__, "iter_next")
                     )
diff --git a/torch/_dynamo/variables/builder.py b/torch/_dynamo/variables/builder.py
index d62dbe6d1f3..d37fcd857a9 100644
--- a/torch/_dynamo/variables/builder.py
+++ b/torch/_dynamo/variables/builder.py
@@ -1349,7 +1349,8 @@ class VariableBuilder:
         self.install_guards(GuardBuilder.RANGE_ITERATOR_MATCH)
         # Get all the values from the range iterator; no need to install guards
         # on items since `RANGE_ITERATOR_MATCH` guarantees the same items.
-        items = [ConstantVariable.create(v) for v in copy.deepcopy(value)]
+        items = [ConstantVariable.create(v) for v in value]
+
         return ListIteratorVariable(items, mutation_type=ValueMutationNew())
 
     def wrap_slice_range(self, value: Union[slice, range]):
