diff --git a/torch/optim/adamw.py b/torch/optim/adamw.py
index 5588a2bc51a..e2352d01a1b 100644
--- a/torch/optim/adamw.py
+++ b/torch/optim/adamw.py
@@ -34,7 +34,10 @@ class AdamW(Adam):
         differentiable: bool = False,
         fused: Optional[bool] = None,
     ):
-        super().__init__(
+        defaults = dict(lr=lr, betas=betas, eps=eps, weight_decay=weight_decay, amsgrad=amsgrad,
+                        maximize=maximize, foreach=foreach, capturable=capturable,
+                        differentiable=differentiable, fused=fused, decoupled_weight_decay=True)
+        super().__init__(params, defaults)
             params,
             lr,
             betas,
